phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKeyIT.java

```diff
@@ -745,6 +744,70 @@ public void testComplexDuplicateKeyExpression() throws Exception {
         }
     }
 
+    @Test
+    public void testRowStampCol() throws Exception {
+        // ROW_TIMESTAMP is not supported for tables with indexes
+        if (indexDDL.length() > 0) {
+            return;
+        }
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            String ddl = "create table " + tableName +
+                "(\n" +
+                "ORGANIZATION_ID CHAR(15) NOT NULL,\n" +
+                "USER_ID CHAR(15) NOT NULL,\n" +
+                "TIME_STAMP DATE NOT NULL,\n" +
+                "STATUS VARCHAR,\n" +
+                "CONSTRAINT PK PRIMARY KEY \n" +
+                "    (\n" +
+                "        ORGANIZATION_ID, \n" +
+                "        USER_ID,\n" +
+                "        TIME_STAMP ROW_TIMESTAMP\n" + // ROW_TIMESTAMP col
+                "    ) \n" +
+                ")\n";
+
+            conn.createStatement().execute(ddl);
+            String orgid = "ORG1";
+            String userid = "USER1";
+            String original = "ORIGINAL";
+            String updated = "UPDATED";
+            String duplicate = "DUPLICATE";
+            long rowTimestamp = EnvironmentEdgeManager.currentTimeMillis() - 10;
```

Comment:
Curious is there a particular reason to subtract 10 ms from the rowTimestamp. ?

Reply:
@ankitjain64  We just wanted a value that is less than the current time. 10 is chosen arbitarily.